src/main/java/de/zalando/aruha/nakadi/repository/db/EventTypeCache.java

```diff
@@ -0,0 +1,140 @@
+package de.zalando.aruha.nakadi.repository.db;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import de.zalando.aruha.nakadi.domain.EventType;
+import de.zalando.aruha.nakadi.exceptions.NoSuchEventTypeException;
+import de.zalando.aruha.nakadi.repository.EventTypeRepository;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.recipes.cache.PathChildrenCache;
+import org.apache.curator.framework.recipes.cache.PathChildrenCacheEvent;
+import org.apache.curator.framework.recipes.cache.PathChildrenCacheListener;
+import org.apache.curator.utils.ZKPaths;
+import org.apache.zookeeper.CreateMode;
+import org.apache.zookeeper.KeeperException;
+
+import java.util.concurrent.ExecutionException;
+
+public class EventTypeCache {
+
+    public final String ZKNODE_PATH = "/nakadi/event_types";
+
+    private final LoadingCache<String, EventType> cache;
+    private final PathChildrenCache cacheSync;
+    private final CuratorFramework zkClient;
+
+    public EventTypeCache(final EventTypeRepository dbRepo, final CuratorFramework zkClient) throws Exception {
+        initParentCacheZNode(zkClient);
+
+        this.zkClient = zkClient;
+        this.cache = setupInMemoryCache(dbRepo);
+        this.cacheSync = setupCacheSync(zkClient);
+    }
+
+    public void updated(final String name) {
+        try {
+            final String path = getZNodePath(name);
+            zkClient.setData().forPath(path, new byte[0]);
```

Comment:
will it actually trigger the watchers if you set the same value as it had before?